packages/suins/sources/suins.move

```diff
@@ -105,6 +108,18 @@ public fun withdraw(
     coin::take(&mut self.balance, amount, ctx)
 }
 
+public fun withdraw_v2<T>(
+    self: &mut SuiNS,
+    _: &AdminCap,
+    ctx: &mut TxContext,
+): Coin<T> {
+    let balance_key = BalanceKey<T> {};
+    assert!(df::exists_(&self.id, balance_key), ENoProfitsInCoinType);
+    let balance: Balance<T> = df::remove(&mut self.id, balance_key);
+
+    balance.into_coin(ctx)
```

Comment:
Let's empty the balance (borrow_mut & extract the amount) so we don't charge an extra DF creation after each withdrawal from users.